### PR TITLE
Use pflag instead of flag in kube2dns

### PR DESCRIPTION
**Pull request by yuvipanda**
- Marks flags with _ as deprecated, use the - variants instead

### CODE REVIEW

**Comment by yuvipanda** [comment]
I can probably also convert these to use a FlagSet if desired.

**Comment by eparis** [comment]
ok to test

**Comment by thockin** [comment]
thanks!

**Comment by thockin** [comment]
We'll need to cut another kube2sky image at some point, but not urgent.

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Comment by wojtek-t** [comment]
@yuvipanda - can you please rebase this PR?

**Comment by wojtek-t** [comment]
@yuvipanda - can you please rebase?

**Comment by gmarek** [comment]
@yuvipanda - please rebase

**Comment by yuvipanda** [comment]
sorry it took a while. done now

On Tue, Jan 12, 2016 at 2:43 PM, k8s-merge-robot <notifications@github.com>
wrote:

> PR changed after LGTM, removing LGTM.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17923#issuecomment-171085137>
> .
>



-- 
Yuvi Panda T
http://yuvi.in/blog

**Comment by ArtfulCoder** [comment]
I will do manual DNS test run with your change today and LGTM it..
code wise it looks good.

**Comment by yuvipanda** [comment]
\o/ awesome, thanks :)

On Thu, Jan 21, 2016 at 2:02 PM, Abhi Shah <notifications@github.com> wrote:

> I will do manual DNS test run with your change today and LGTM it..
> code wise it looks good.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17923#issuecomment-173722815>
> .
>



-- 
Yuvi Panda T
http://yuvi.in/blog

**Comment by ArtfulCoder** [comment]
I had to update  k8s.io/kubernetes/cluster/addons/dns/skydns-rc.yaml.in and change 
-domain to --domain for the kube2sky container argument.

Can you update your PR with this file change.

**Comment by yuvipanda** [comment]
Done!

**Comment by ArtfulCoder** [comment]
apparently, you need a rebase again...

**Comment by ArtfulCoder** [comment]
I have LGTMed the PR though. thank you for doing it!

**Comment by yuvipanda** [comment]
I've rebased and pushed! \o/ Thanks for reviewing :)

On Thu, Jan 21, 2016 at 4:26 PM, Abhi Shah <notifications@github.com> wrote:

> I have LGTMed the PR though. thank you for doing it!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17923#issuecomment-173760033>
> .
>



-- 
Yuvi Panda T
http://yuvi.in/blog

**Comment by ArtfulCoder** [comment]
can you run:
 hack/verify-flags-underscore.py -e > hack/verify-flags/exceptions.txt` to update the list

**Comment by yuvipanda** [comment]
Done!

On Thu, Jan 21, 2016 at 10:41 PM, Abhi Shah <notifications@github.com>
wrote:

> can you run:
> hack/verify-flags-underscore.py -e > hack/verify-flags/exceptions.txt` to
> update the list
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17923#issuecomment-173825358>
> .
>



-- 
Yuvi Panda T
http://yuvi.in/blog

**Comment by ArtfulCoder** [comment]
unit tests are passing now
lets wait for an e2e run :)

**Comment by eparis** [comment]
@k8s-bot e2e test this

File: hack/verify-flags/known-flags.txt [hunk 1]
```diff
@@ -174,6 +176,8 @@ kubelet-read-only-port
 kubelet-root-dir
 kubelet-sync-frequency
 kubelet-timeout
+kube-master
```
Comment (eparis) [inline hack/verify-flags/known-flags.txt]: Why is `kube-master` listed twice?

**Pull request merged**